### PR TITLE
[COPS-3222][DCOS-38688] Ensure Mounts for TLS secrets are not duplicate

### PR DESCRIPTION
**Pull request by elezar**
This PR is a backport of #2556 and #2576.

This PR addresses an issue where we request multiple mounts of the same secret on TLS enabled data services. This currently only affects HDFS, but could be an issue for any service running multiple tasks with TLS per pod.

The changes included are:
* Modify `TLSEvaluationStage` to not add duplicate volumes to the `PodInfoBuilder`
* Modify `OfferEvaluator` to not add duplicate `TLSEvaluationStages` to the offer evaluation pipeline
* Add `pod replace` integration tests for HDFS, Kafka, Elastic, and Cassandra

(The tests were added first to confirm that this was only affecting HDFS -- after which the fix was implemented).

*Note*: I had originally tried to also run this through a `ServiceTestRunner`, but found that the TLS-handling (especially the Secrets Client) is a bit tightly coupled and I did not want to refactor this further.

### Bugfix
* Fix an error with duplicate mount operations on TLS-enabled HDFS

### CODE REVIEW

File: frameworks/hdfs/tests/test_sanity.py [hunk 1]
```diff
@@ -380,18 +374,29 @@ def expected_metrics_exist(emitted_metrics):
...
-    data_ids = sdk_tasks.get_task_ids(foldered_name, 'data')
 
-    sdk_cmd.svc_cli(config.PACKAGE_NAME, foldered_name, 'pod replace {}'.format(name_node_name))
+    pod_list = ["name-0", "name-1", "name-0"]
+    for pod in pod_list:
+        sdk_recovery.check_permanent_recovery(config.PACKAGE_NAME,
+                                              foldered_name,
+                                              pod,
+                                              recovery_timeout_s=25 * 60
+                                              )
```
Comment (mpereira) [inline frameworks/hdfs/tests/test_sanity.py]: Weird formatting.

File: frameworks/hdfs/tests/test_sanity.py [hunk 2]
```diff
@@ -380,18 +374,29 @@ def expected_metrics_exist(emitted_metrics):
...
+def test_permanently_replace_journalnodes():
+    foldered_name = sdk_utils.get_foldered_name(config.SERVICE_NAME)
+
+    pod_list = ["journal-0", "journal-1", "journal-2"]
+    for pod in pod_list:
+        sdk_recovery.check_permanent_recovery(config.PACKAGE_NAME,
+                                              foldered_name,
+                                              pod,
+                                              recovery_timeout_s=25 * 60
+                                              )
```
Comment (mpereira) [inline frameworks/hdfs/tests/test_sanity.py]: Weird formatting.

Comment (elezar) [reply frameworks/hdfs/tests/test_sanity.py]: 914979f

File: frameworks/hdfs/tests/test_sanity.py (hunk 1 above)
Comment (elezar) [reply frameworks/hdfs/tests/test_sanity.py]: 914979f

**Comment by elezar** [comment]
@mpereira have you had a chance to look into the Java changes?

**Pull request merged**